Sweave2rmd: BiobaseDevelopment.Rnw to Rmd

hsadia538 opened the pull request:
Part of Sweave2rmd project.

- Creates a vignette in R markdown for BiobaseDevelopment.Rnw file which replaces the sweave vignette.
- Fix spelling errors

Sample vignettes are attached below:

[BiobaseDevelopment Rmd File](https://github.com/Bioconductor/Biobase/files/9771211/Biobase.development.and.the.new.eSet.html.zip)
[BiobaseDevelopment Rnw File](https://bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/BiobaseDevelopment.pdf)

Commit:
Sweave2rmd: BiobaseDevelopment.Rnw to Rmd

-Convert BiobaseDevelopment.Rnw to Rmd file
- Fix spelling erros

hsadia538 commented:
Hi @jwokaty, Can you please take a look? 
Let me know incase changes are needed.Thanks

Commit:
Remove BiobaseDevelopment.Rmd

Commit:
Add BiobaseDevelopment.Rmd

jwokaty commented:
@mcarlsn Can you do the initial review?

mcarlsn commented on vignettes/BiobaseDevelopment.Rmd:
```diff
@@ -0,0 +1,504 @@
+---
+title: "*Biobase* development and the new `eSet`"
+author: R. Gentleman, V. Carey, M. Morgan, S. Falcon
```
The author line in this file and the vignette PDF don't quite match up. In the PDF, the author is Martin T. Morgan. Please update the author line to match the PDF (see link: https://bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/BiobaseDevelopment.pdf)

mcarlsn commented on vignettes/BiobaseDevelopment.Rmd:
Small edit here: in the PDF, this blurb does not have a bullet point. Please remove the bullet point dash. Bullet point list will start again at "title: experiment title."

mcarlsn commented on vignettes/BiobaseDevelopment.Rmd:
Another small edit: `preprocessing` : `list` should be a joined line of code instead of separate syntax words separated by a colon. Please change to `preprocessing: list` like so: 
<img width="148" alt="Screen Shot 2022-10-19 at 4 04 49 PM" src="https://user-images.githubusercontent.com/105299892/196792729-8ff8553b-6bbe-42d6-8b11-d22c6baf3090.png">

mcarlsn commented on vignettes/BiobaseDevelopment.Rmd:
```diff
@@ -0,0 +1,504 @@
+date:  "`r format(Sys.time(), '%d %B, %Y')`"
+package: Biobase
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
```
Please update "Vignette Title" to "Biobase-BiobaseDevelopment"

mcarlsn commented on vignettes/BiobaseDevelopment.Rmd:
Small edit: In this passage, you write `R, G, Rb, Rg` but above, in a similar list, you formatted it as `R`, `G`, `Rb`, `Gb`

For the sake of consistency, please select one formatting style and apply it to both lists.

mcarlsn commented on vignettes/BiobaseDevelopment.Rmd:
To better match the formatting in the PDF, please change sessionInfo() to 
<img width="607" alt="Screen Shot 2022-10-20 at 3 59 44 PM" src="https://user-images.githubusercontent.com/105299892/197045982-35e665e6-cd83-4e38-a6eb-a83994815bf8.png">

mcarlsn replied:
Which looks like this when knitted: 
<img width="779" alt="Screen Shot 2022-10-20 at 4 00 18 PM" src="https://user-images.githubusercontent.com/105299892/197046097-d1122c20-37e5-444a-9af6-bc7064ed2b97.png">

hsadia538 replied:
You mean hardcoding the session info here?

jwokaty replied:
We shouldn't hard code this.

hsadia538 replied:
Should I revert back to `sessionInfo()`

mcarlsn replied:
Apologies for the misdirection @hsadia538! I went back and forth on this edit before commenting - glad to know we should stick to sessionInfo(). Hopefully it wasn't too much of a hassle for you to revert this edit.

hsadia538 replied:
No worries.

hsadia538 replied:
I have updated. Please take a look thanks 🙂

mcarlsn reviewed (changes requested):
Wonderful work on this conversion @hsadia538. 👏🏻👏🏻 I left a few comments for small changes/ edits. @jwokaty FYI there is something interesting going on with the syntax in this vignette (not because of anything you did, hsadia538!). The code runs fine when I go chunk by chunk, but it runs into errors aplenty when knitted. All errors seem tied to `eSet` and `SwirlSet` & the error messages are all variations of "Error in getClass("eSet") : "eSet" is not a defined class". These are the following lines that produce errors when knitted: 
- 65
- 211
- 269 
- 289-290
- 343 
- 399-404
- 421-445
- 451
- 461
- 471-475
- 483-485
- 489
Any ideas @jwokaty? I'm going to keep playing w/ it. & Thanks again, @hsadia538!

hsadia538 commented:
>   Any ideas @jwokaty? I'm going to keep playing w/ it. & Thanks again, @hsadia538!

Thank you for the detailed review. I will incorporate the feedback and push changes.

One thing I did before knitting was using `devtools::load_all(".")` at the start which helped me knit and verify the output. I removed it later on as its not part of the code.

Commit:
Update BiobaseDevelopment.Rmd

hsadia538 commented:
@mcarlsn I have pushed changes. Please take a look. thanks

jwokaty commented:
@hsadia538 @mcarlsn 

> Wonderful work on this conversion @hsadia538. 👏🏻👏🏻 I left a few comments for small changes/ edits. @jwokaty FYI there is something interesting going on with the syntax in this vignette (not because of anything you did, hsadia538!). The code runs fine when I go chunk by chunk, but it runs into errors aplenty when knitted. All errors seem tied to `eSet` and `SwirlSet` & the error messages are all variations of "Error in getClass("eSet") : "eSet" is not a defined class". These are the following lines that produce errors when knitted:
> 
> * 65
> * 211
> * 269
> * 289-290
> * 343
> * 399-404
> * 421-445
> * 451
> * 461
> * 471-475
> * 483-485
> * 489
>   Any ideas @jwokaty? I'm going to keep playing w/ it. & Thanks again, @hsadia538!

Usually when this happens it means we're missing some code from the original. Typically, we should see a statement loading the package the vignette belongs to with `library()`. In the Rnw, we have https://github.com/Bioconductor/Biobase/blob/6fd3f136c23c8028d6340f5aabddf64dfc6ba809/vignettes/BiobaseDevelopment.Rnw#L100-L103

So we should add after "# A quick tour"

    ```{r echo=FALSE, message=FALSE}
    options(width=69)
    library(Biobase)
    ```
Then we don't see the errors.

Commit:
Update BiobaseDevelopment

Commit:
Merge branch 'master' into BiobaseDevelopment-rmd

Commit:
Update BiobaseDevelopment.Rmd

Commit:
Update BiobaseDevelopment.Rmd

- Based on latest feedback on esApply.Rmd PR

jwokaty commented on DESCRIPTION:
```diff
@@ -17,7 +17,7 @@ Maintainer: Bioconductor Package Maintainer
 Depends: R (>= 2.10), BiocGenerics (>= 0.27.1), utils
 Imports: methods
-Suggests: tools, tkWidgets, ALL, RUnit, golubEsets, BiocStyle, knitr
+Suggests: tools, tkWidgets, ALL, RUnit, golubEsets, BiocStyle, Knitr 
```
Since we already made all the changes for the `DESCRIPTION` file in #17, let's remove this change.

jwokaty replied:
We still need the `Suggests` line, but this diff shows that the `k` in `knitr` was capitalized, so we just needed to undo that.

jwokaty commented on vignettes/BiobaseDevelopment.Rmd:
```diff
@@ -0,0 +1,510 @@
+---
+title: "*Biobase* development and the new `eSet`"
+author:
+- name "Martin T. Morgan*"
```
Let's remove the `*` after `Morgan`. The email that's displayed in the PDF is old, so we don't want to use that.  Also add the `:` after `name`.

jwokaty commented on vignettes/BiobaseDevelopment.Rmd:
Can we format for easier reading:

```
obj <- new(
    "ExpressionSet", 
    phenoData = new("AnnotatedDataFrame"),
    experimentData = new("MIAME"), annotation = character(), 
    exprs = new("matrix")
)
```

jwokaty commented on vignettes/BiobaseDevelopment.Rmd:
After the `)` on line 321, we should have a space followed by `{` because it's the start of a function.

The 2nd `)` on 324 should be on the next line with the closing `}` before it--e.g, `})`.

jwokaty commented on vignettes/BiobaseDevelopment.Rmd:
Let's fix the indent of `warning` and the closing `}`.

jwokaty commented on vignettes/BiobaseDevelopment.Rmd:
I think we need `eval=FALSE` to prevent the code from being run.

jwokaty commented on vignettes/BiobaseDevelopment.Rmd:
Let's make `BiocGenerics` a link to the package with `Biocpkg("BiocGenerics")`

hsadia538 replied:
Do you want me to adda hyperlink and link it to package installation page?

jwokaty replied:
Since we're using BiocStyle, the function `Biocpkg()` will make the link for us. (Sorry that wasn't clear!)

hsadia538 replied:
Ok got it , thanks
Also one for thing, I have made multiple commits here. Is it okay as is or I should squash them to keep the history clean?

hsadia538 replied:
The function knits as followes:
`(the updateObject generic function is defined in the Biocpkg(“BiocGenerics”) package) for conveniently updating old objects to their new representation.`

Its not turning into a link

jwokaty replied:
I will squash commits when I merge.  For the link, try using `r Biocpkg("BiocGenerics")`. Here's the BiocStyle documentation: https://bioconductor.org/packages/release/bioc/vignettes/BiocStyle/inst/doc/AuthoringRmdVignettes.html#4_Style_macros.

hsadia538 replied:
Done and pushed changes.
Thank you

jwokaty commented on vignettes/BiobaseDevelopment.Rmd:
Remove extra line 62 and let's remove line 60 `options(width=69)`.

jwokaty commented on vignettes/BiobaseDevelopment.Rmd:
We should use the code block structure for this with `eval=FALSE` so that we can remove the `>`.

jwokaty reviewed (changes requested):
Thanks for working on this vignette. This is just my way of thinking but maybe we don't need the spaces between each list item since it makes the document long for future vignettes; we'll leave them for this vignette. Can I ask you to make the other changes and re-request a review when they're done?

hsadia538 commented:
> Thanks for working on this vignette. This is just my way of thinking but maybe we don't need the spaces between each list item since it makes the document long for future vignettes; we'll leave them for this vignette. Can I ask you to make the other changes and re-request a review when they're done?

Hi, which list items are you exactly referring to(just for understanding purposes)

Commit:
update

jwokaty commented:
> > Thanks for working on this vignette. This is just my way of thinking but maybe we don't need the spaces between each list item since it makes the document long for future vignettes; we'll leave them for this vignette. Can I ask you to make the other changes and re-request a review when they're done?
> 
> Hi, which list items are you exactly referring to(just for understanding purposes)

Thanks for asking for clarification. I was referring to lines like 43-55, where there's a blank line after each item in the list. As I mentioned, let's leave them for now.

Commit:
Update BiobaseDevelopment.Rmd

Commit:
Merge branch 'master' into BiobaseDevelopment-rmd

jwokaty committed:
Fix DESCRIPTION, edit BiobaseDevelopment.Rmd

- Add Suggests
- Format BiobaseDevelopment, fix code blocks, remove blank lines

jwokaty reviewed (approved):
I made a few changes to get this ready for merge, but I wanted to mention a few things. The `Suggest` line in DESCRIPTION needed to be added back. `Authors@R`, is a vector of lists, so the commas are important. The remainder of changes were primarily formatting issues to improve code blocks.
